packages/beacon-node/src/db/repositories/dataColumnSidecar.ts

```diff
@@ -27,21 +29,21 @@ export class DataColumnSidecarRepository extends PrefixedRepository<BlockRoot, C
   }
 
   encodeKeyRaw(prefix: BlockRoot, id: ColumnIndex): Uint8Array {
-    return Buffer.concat([prefix, intToBytes(id, 4)]);
+    return Buffer.concat([prefix, encodeNumberForDbKey(id, columnIndexByteSize)]);
   }
 
   decodeKeyRaw(raw: Uint8Array): {prefix: BlockRoot; id: ColumnIndex} {
     return {
-      prefix: raw.slice(0, 32) as BlockRoot,
-      id: bytesToInt(raw.slice(32, 36)) as ColumnIndex,
+      prefix: raw.slice(0, blockRootByteSize) as BlockRoot,
+      id: decodeNumberForDbKey(raw.slice(blockRootByteSize), columnIndexByteSize) as ColumnIndex,
     };
   }
 
   getMaxKeyRaw(prefix: BlockRoot): Uint8Array {
```

Comment:
We need to ensure that usage of this method doesn't assume exclusive, which this PR is changing.

Reply:
Yes that also verified, we only use that function in the abstract class. 

<img width="473" height="489" alt="image" src="https://github.com/user-attachments/assets/09b8b9e2-d546-4633-b12d-5494acbd7989" />

Reply:
I think https://github.com/ChainSafe/lodestar/pull/8357 added one. May need to merge unstable back here